### DISCUSSION TITLE
How to co-locate Stimulus controllers and CSS modules?

### DISCUSSION THREAD

**Discussion by RomanTurner** [ElMassimo/vite_ruby#371] [General]
Thanks for this wonderful project. It has been great working with vite_ruby. 

I am using vite_ruby with roda, the web toolkit, to make a custom framework that aligns around the locality of behavior. 
From the code to the way the code is organized, everything that goes together lives together. 

example directory structure: 
```ruby
project
│   README.md
│   app.rb
|   config.ru
│
└───allocs
│   │
│   └─── users
│       │   data_model.rb
│       │   routes.rb
│       └─── components
│                   └ user_card
│                       │   user_card.rb
│                       │   user_card_controller.js
│                       │   user_card.module.css
```
I would love to be able to add stimulus controllers and CSS modules inside of the allocated resource files. Would there be a way to glob search the app directory for the js/css files and build them from there?

**Comment by ElMassimo** [comment]
Hi Roman!

The following section of the guide discusses a few options:

- [Bundling files outside sourceCodeDir](https://vite-ruby.netlify.app/guide/advanced.html#bundling-files-outside-sourcecodedir)

Which style to use depends mostly on the trade-offs of bundling all the globbed components together, or bundling each section as a separate chunk.

**Reply by RomanTurner** [reply]
> Hi Roman!
> 
> 
> 
> The following section of the guide discusses a few options:
> 
> 
> 
> - [Bundling files outside sourceCodeDir](https://vite-ruby.netlify.app/guide/advanced.html#bundling-files-outside-sourcecodedir)
> 
> 
> 
> Which style to use depends mostly on the trade-offs of bundling all the globbed components together, or bundling each section as a separate chunk.

Thank you for the quick reply! My eyes must have missed this section. Thank you again.
